m2/src/classes/Product.js

```diff
@@ -0,0 +1,53 @@
+export class Product {
+    #name;
+    #description;
+    #price;
+    #tags;
+    #images;
+    #favoriteCount;
+
+    constructor({name, description, price, tags = [], images = [], favoriteCount = 0}) {
+        this.#name = name; // 상품명
+        this.#description = description; // 상품 설명
+        this.#price = price; // 판매 가격
+        this.#tags = tags; // 해시태그 배열
+        this.#images = images; // 이미지 배열
```

Comment:
복사본으로 넣어주세요. 참조타입은 원본이 변하면 내부 멤버변수가 같이 변합니다.
```suggestion
        this.#tags = [...tags]; // 해시태그 배열
        this.#images = [...images]; // 이미지 배열
```